flight/odlc_boundaries/search_path.py

```diff
@@ -0,0 +1,204 @@
+        coords["latitude"], coords["longitude"]
+    )
+    coords["utm_x"] = utm_coords[0]
+    coords["utm_y"] = utm_coords[1]
+    return coords
+
+
+def all_latlon_to_utm(list_of_coords: list[dict[str, float]]) -> list[dict[str, float]]:
+    """
+    Converts a list of dictionaries with latlon data to add utm data
+
+    Parameters
+    ----------
+    list_of_coords : list[dict[str, union[float, int, str]]]
+        A list of dictionaries that contain lat long data
+
+    Returns
+    -------
+    list_of_coords : [dict[str, union[float, int, str]]]
+        An updated list of dictionaries with added utm data
+    """
+    coord: int
+    for coord, _ in enumerate(list_of_coords):
+        list_of_coords[coord] = latlon_to_utm(list_of_coords[coord])
+    return list_of_coords
+
+
+def generate_search_paths(
+    d_search_area_boundary: list[dict[str, float]], buffer_distance: int
+) -> list[tuple[float, float]]:
+    """
+    Generates a list of search paths of increasingly smaller sizes until the whole area
+    of the original shape has been covered
+
+    Parameters
+    ----------
+    d_search_area_boundary : List[Dict[str, float]]]
+        A list of coordinates in dictionary form that contain utm coordinate data
+    buffer_distance : int
+        The distance that each search path will be apart from the previous one.
+        For complete photographic coverage of the area, this should be equal to half the height
+        of the area the camera covers on the ground given the current altitude.
+
+    Returns
+    -------
+    generated_search_paths : [Tuple[float, float]]
+        A list of concentric search paths that cover the area of the polygon
+    """
+
+    # convert to shapely polygon for buffer operations
+    search_area_points: list[tuple[float, float]] = [
+        (point["utm_x"], point["utm_y"]) for point in d_search_area_boundary
+    ]
+    boundary_shape: Polygon = Polygon(search_area_points)
+
+    generated_search_paths: list[tuple[float, float]] = []
+    # shrink boundary by a fixed amount until the area it covers is 0
+    # add the smaller boundary to our list of search paths on each iteration
+    while boundary_shape.area > 0:
+        # print(boundary_shape.exterior.coords.xy)
+        generated_search_paths.append(
+            tuple(zip(*boundary_shape.exterior.coords.xy))  # pylint: disable=maybe-no-member
+        )
+        boundary_shape = boundary_shape.buffer(buffer_distance, single_sided=True)
+
+    return generated_search_paths
+
+
+async def run() -> None:
+    """
+    This function is just a driver to test the goto function and runs through the
+    entire waypoint section of the SUAS competition
+    """
+
+    waypoint: dict[str, list[float]] = {
+        "lats": [38.31451966813249, 38.31430872867596, 38.31461622313521],
+        "longs": [-76.54519982319357, -76.54397320409971, -76.54516993186949],
+        "Altitude": [85],
+    }
```

Comment:
This function appears to be some sort of test function. I'm not sure where the search path should be called from though, so check in with @cjhr95 .

Reply:
The waypoint data is currently just test data, however, for the competition, this part of the code will be run first when starting the Air Drop sequence, and the waypoints here will be the midpoints of the edges of the rectangle to search.